src/sqlodbc/opensearch_communication.h

```diff
@@ -82,13 +81,17 @@ class OpenSearchCommunication {
                          ConnErrorType error_type);
     void SetErrorDetails(ErrorDetails details);
 
+    void SetIsServerless();
+    void SetSqlEndpoint();
+
     // TODO #35 - Go through and add error messages on exit conditions
     std::string m_error_message;
     const std::vector< std::string > m_supported_client_encodings = {"UTF8"};
 
     ConnStatusType m_status;
     ConnErrorType m_error_type;
     std::shared_ptr< ErrorDetails > m_error_details;
+    bool is_serverless;
```

Comment:
why not is_aoss_serverless?

Reply:
I don't feel strongly about it. Serverless only exists as part of Amazon OpenSearch Service, so the "aoss" doesn't necessary add anything, but not a bad idea to be more explicit/tie it more clearly to the `aoss.amazonaws.com` URL component. 🤷

Reply:
Always be explicit where possible.

Reply:
Updated to `is_aoss_serverless`. Also updated corresponding variable, constant, and function names for consistency.